src/libcprover-cpp/api.cpp

```diff
@@ -47,8 +42,58 @@ api_sessiont::api_sessiont(const api_optionst &options)
 
 api_sessiont::~api_sessiont() = default;
 
-void api_sessiont::load_model_from_files(const std::vector<std::string> &files)
+struct api_messaget
 {
+  std::string string;
+};
+
+const char *api_message_get_string(const api_messaget &message)
+{
+  return message.string.c_str();
+}
+
+class api_message_handlert : public message_handlert
+{
+public:
+  explicit api_message_handlert(
```

Comment:
Is the callback handled synchronously or async?

Reply:
I'm assuming sync, but probably good to document

Reply:
Its handled in a single threaded fashion. So the process flow would be something like -
1) API is called.
2) CProver processing happens.
3) Message callback is issued.
4) Callback processes the message.
5) Callback returns.
6) CProver processing continues.
7) CProver processing completes and API call from step 1 returns.